### PR TITLE
[Enhancement] (nereids)implement showDiagnoseTabletCommand in nereids

### LINKED ISSUE #42836: [Enhancement] (nereids)implement showDiagnoseTabletCommand in nereids
showDiagnoseTablet command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showDiagnoseTablet command in nereids:
unsupportedshowDiagnoseTabletStatement
``` SHOW TABLET DIAGNOSIS tabletId=INTEGER_VALUE                                  ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42836 

implement showDiagnoseTabletCommand in nereids

2024-11-20 15:03:26.592 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: CREATE TABLE IF NOT EXISTS test_table_for_diagnosis (id INT, name STRING) DUPLICATE KEY(id) DISTRIBUTED BY HASH(id) BUCKETS 3 PROPERTIES('replication_num'='1');
2024-11-20 15:03:26.673 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: SHOW TABLETS FROM test_table_for_diagnosis
2024-11-20 15:03:26.717 INFO [suite-thread-1] (Suite.groovy:717) - start check5f657d91-8875-4700-b3f3-56071b69ea34
2024-11-20 15:03:26.731 INFO [suite-thread-1] (Suite.groovy:721) - start profile run5f657d91-8875-4700-b3f3-56071b69ea34
2024-11-20 15:03:26.734 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: --5f657d91-8875-4700-b3f3-56071b69ea34
SHOW TABLET DIAGNOSIS 11006
2024-11-20 15:03:26.987 INFO [suite-thread-1] (HttpCliAction.groovy:106) - url : http://127.0.0.1:8030/rest/v1/query_profile
2024-11-20 15:03:26.988 INFO [suite-thread-1] (HttpCliAction.groovy:107) - body: null
2024-11-20 15:03:26.989 INFO [suite-thread-1] (HttpCliAction.groovy:108) - op: get
log4j:WARN No appenders could be found for logger (org.apache.http.client.protocol.RequestAddCookies).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
2024-11-20 15:03:27.294 INFO [suite-thread-1] (HttpCliAction.groovy:106) - url : http://127.0.0.1:8030/rest/v1/query_profile/30676fd76b7c42a6-8a9c53747e26a1cf
2024-11-20 15:03:27.295 INFO [suite-thread-1] (HttpCliAction.groovy:107) - body: null
2024-11-20 15:03:27.296 INFO [suite-thread-1] (HttpCliAction.groovy:108) - op: get
2024-11-20 15:03:27.338 INFO [suite-thread-1] (Suite.groovy:726) - start profile check5f657d91-8875-4700-b3f3-56071b69ea34
2024-11-20 15:03:27.339 INFO [suite-thread-1] (Suite.groovy:727) - Summary:
      -  Profile  ID:  30676fd76b7c42a6-8a9c53747e26a1cf
      -  Task  Type:  QUERY
      -  Start  Time:  2024-11-20  15:03:26
      -  End  Time:  2024-11-20  15:03:26
      -  Total:  2ms
      -  Task  State:  EOF
      -  User:  root
      -  Default  Catalog:  internal
      -  Default  Db:  regression_test_nereids_p0_show
      -  Sql  Statement:  --5f657d91-8875-4700-b3f3-56071b69ea34
SHOW  TABLET  DIAGNOSIS  11006
      -  Distributed  Plan:  N/A
Execution  Summary:
      -  Parse  SQL  Time:  1ms
      -  Nereids  Analysis  Time:  N/A
      -  Nereids  Rewrite  Time:  N/A
      -  Nereids  Optimize  Time:  N/A
      -  Nereids  Translate  Time:  N/A
      -  Nereids  Distribute  Time:  N/A
      -  Workload  Group:  
      -  Analysis  Time:  N/A
      -  Plan  Time:  N/A
          -  JoinReorder  Time:  N/A
          -  CreateSingleNode  Time:  N/A
          -  QueryDistributed  Time:  N/A
          -  Init  Scan  Node  Time:  N/A
          -  Finalize  Scan  Node  Time:  N/A
              -  Get  Splits  Time:  N/A
                  -  Get  Partitions  Time:  N/A
                  -  Get  Partition  Files  Time:  N/A
              -  Create  Scan  Range  Time:  N/A
          -  Get  Partition  Version  Time:  N/A
          -  Get  Partition  Version  Count  (hasData):  N/A
          -  Get  Partition  Version  Count:  N/A
          -  Get  Table  Version  Time:  N/A
          -  Get  Table  Version  Count:  N/A
      -  Schedule  Time:  N/A
          -  Fragment  Assign  Time:  N/A
          -  Fragment  Serialize  Time:  N/A
          -  Fragment  RPC  Phase1  Time:  N/A
          -  Fragment  RPC  Phase2  Time:  N/A
          -  Fragment  Compressed  Size:  0.00  
          -  Fragment  RPC  Count:  0
      -  Schedule  Time  Of  BE:  {}
      -  Wait  and  Fetch  Result  Time:  N/A
      -  Fetch  Result  Time:  0ms
      -  Write  Result  Time:  0ms
      -  Doris  Version:  6898b77507
      -  Is  Nereids:  Yes
      -  Is  Cached:  No
      -  Total  Instances  Num:  0
      -  Instances  Num  Per  BE:  
      -  Parallel  Fragment  Exec  Instance  Num:  4
      -  Trace  ID:  
      -  Transaction  Commit  Time:  N/A
      -  System  Message:  N/A
      -  Executed  By  Frontend:  N/A
      -  Nereids  GarbageCollect  Time:  -1ms
      -  Nereids  BeFoldConst  Time:  0ms

Changed  Session  Variables:
VarName                              |  CurrentValue                |  DefaultValue
----------------------|---------------------|-------------
character_set_results  |  NULL                                |  utf8mb4          
sql_mode                            |  STRICT_TRANS_TABLES  |                          
variable_version            |  0                                      |  300                  

2024-11-20 15:03:27.375 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: DROP TABLE IF EXISTS test_table_for_diagnosis
2024-11-20 15:03:27.404 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_show_diagnose_tablet_nereids in /root/doris/doris/regression-test/suites/nereids_p0/show/test_show_diagnose_tablet.groovy succeed
2024-11-20 15:03:46.547 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-20 15:05:06.787 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/doris/regression-test/suites/nereids_p0/show/test_show_diagnose_tablet.groovy: group=query,diagnose,p0, name=test_show_diagnose_tablet_nereids
2024-11-20 15:05:06.789 INFO [main] (RegressionTest.groovy:492) - All suites success.

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.apache.doris.system.Diagnoser;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * Represents the command for SHOW TABLET DIAGNOSIS
+ */
+public class ShowDiagnoseTabletCommand extends Command implements NoForward {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java:42]:
```suggestion
public class ShowDiagnoseTabletCommand extends ShowCommand {
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowDiagnoseTabletCommand(this, context);
+    }
+
+    @Override
+    public StmtType stmtType() {
+        return StmtType.SHOW;
+    }
+}
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java:84]: override toRedirectStatus, refer to DiagnoseTabletStmt

**Comment by starocean999** [comment]
please add test case

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java (hunk 1 above)
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowDiagnoseTabletCommand.java]: done - please review

**Comment by msridhar78** [comment]
> please add test case

Added

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**